app/components/Popover/index.js

```diff
@@ -2,82 +2,47 @@
 
 import type { Node } from 'react';
 
-import { Component } from 'react';
+import { useRef, useState } from 'react';
 import { Overlay } from 'react-overlays';
 import cx from 'classnames';
 import styles from './Popover.css';
 
 type Props = {
-  render: () => Node,
+  triggerComponent: Node,
   children: any,
-  placement: 'top' | 'bottom' | 'left' | 'right',
   contentClassName?: string,
 };
 
-type State = {
-  contentHovered: boolean,
-  overlayHovered: boolean,
-};
-
-class Popover extends Component<Props, State> {
-  state = {
-    contentHovered: false,
-    overlayHovered: false,
-  };
-
-  target: any;
-
-  static defaultProps = {
-    placement: 'bottom',
-  };
-
-  onMouseEnterContent = () => {
-    this.setState({ contentHovered: true });
-  };
-
-  onMouseLeaveContent = () => {
-    this.setState({ contentHovered: false });
-  };
-
-  onMouseEnterOverlay = () => {
-    this.setState({ overlayHovered: true });
-  };
-
-  onMouseLeaveOverlay = () => {
-    this.setState({ overlayHovered: false });
-  };
-
-  showOverlay = () => {
-    return this.state.contentHovered || this.state.overlayHovered;
-  };
-
-  render() {
-    const { children, placement, contentClassName } = this.props;
-
-    return (
-      <div
-        onMouseEnter={this.onMouseEnterContent}
-        onMouseLeave={this.onMouseLeaveContent}
-        ref={(target) => {
-          this.target = target;
-        }}
-      >
-        {this.props.render && this.props.render()}
-
-        {this.showOverlay() && (
-          <Overlay show placement={placement} target={this.target}>
+const Popover = ({ triggerComponent, children, contentClassName }: Props) => {
+  const [contentHovered, setContentHovered] = useState(false);
+  const [overlayHovered, setOverlayHovered] = useState(false);
+  const targetRef = useRef(null);
```

Comment:
We should consistently type these, and I'm surprised flow isn't bitching about it.

Reply:
I assume you're talking about the `useState`s. I will add types, but I think flow might be able to imply type when you provide a default value.